client/src/portal/mod.rs

```diff
@@ -210,6 +210,49 @@ impl Keyring {
         }
     }
 
+    /// Verify the secret associated with a keyring.
+    ///
+    /// This function supports all keyring formats (legacy and latest).
+    ///
+    /// # Arguments
+    ///
+    /// *`name` - The name of the keyring.
+    /// * `secret` - The service key, usually retrieved from the Secrets portal.
+    pub async fn verify_keyring_secret(name: &str, secret: Secret) -> bool {
+        #[cfg(feature = "tracing")]
+        tracing::debug!("Trying to verify {} keyring secret", name);
+
+        let keyring = match Keyring::open(name, secret).await {
+            Ok(keyring) => keyring,
+            Err(_err) => {
+                #[cfg(feature = "tracing")]
+                tracing::debug!("Wrong secret for {}: {}", name, _err);
+                return false;
+            }
+        };
+
+        let mut ret = false;
+
+        let items = keyring.items().await;
+        for item in items {
+            match item {
+                Ok(_) => {
+                    #[cfg(feature = "tracing")]
+                    tracing::debug!("{} keyring secret matches", name);
+                    ret = true;
+                    break;
```

Comment:
change this to `return true`. And remove the `let mut ret`, you can just return `false` at the end.

Reply:
got it, thanks @A6GibKm for the prompt review!